libfuse/dir.go

```diff
@@ -334,6 +342,33 @@ func (f *Folder) tlfHandleChangeInvalidate(ctx context.Context,
 	}
 }
 
+// fillAttrWithUIDAndWritePerm sets attributes based on the entry info, and
+// pops in correct UID and write permissions. It only handles fields common to
+// all entryinfo types.
+func (f *Folder) fillAttrWithUIDAndWritePerm(
+	ctx context.Context, ei *libkbfs.EntryInfo, a *fuse.Attr) {
+	a.Valid = 1 * time.Minute
+
+	a.Size = ei.Size
+	a.Mtime = time.Unix(0, ei.Mtime)
+	a.Ctime = time.Unix(0, ei.Ctime)
+
+	a.Uid = uint32(os.Getuid())
+
+	a.Mode &^= os.FileMode(0222) // clear write perm bits
+	_, uid, err := libkbfs.GetCurrentUserInfoIfPossible(
+		ctx, f.fs.config.KBPKI(), f.list.public)
+	// Here we get an error, but there is little that can be done.
+	// cuser will be empty in the error case in which case we will default to the
+	// canonical format.
+	if err != nil {
```

Comment:
We shouldn't ignore the error, because it's only non-`nil` if a real problem happened.  We should return it to the caller.  If the user is logged out, we will get an empty username and uid, with a `nil` error.